src/main/java/org/mule/extension/ftp/internal/config/FileConnectorConfig.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.ftp.internal.config;
+
+import org.mule.extension.ftp.internal.connection.FileSystem;
+import org.mule.runtime.extension.api.annotation.param.DefaultEncoding;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.RefName;
+import org.mule.runtime.extension.api.annotation.param.display.DisplayName;
+import org.mule.runtime.extension.api.annotation.param.display.Placement;
+import org.mule.runtime.extension.api.annotation.param.display.Summary;
+
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+import static org.mule.runtime.extension.api.annotation.param.display.Placement.ADVANCED_TAB;
+
+/**
+ * Generic contract for a config of a connector which operates over a {@link FileSystem}
+ *
+ * @since 1.0
+ */
+public abstract class FileConnectorConfig {
+
+  @RefName
+  private String configName;
+
+  @DefaultEncoding
+  private String muleEncoding;
+
+  /**
+   * Deprecated parameter
+   */
+  @Parameter
+  @Summary("This parameter is deprecated and will not be taken into account")
+  @DisplayName("Default write encoding(DEPRECATED)")
+  @Optional
+  @Placement(tab = ADVANCED_TAB)
+  private String defaultWriteEncoding;
```

Comment:
If used anywhere, review. In any case, since it is deprecated, this needs to be removed.

Reply:
Included it as part of refactoring user story : https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001xkdp5YAA/view

Reply:
Once more, removal of deprecated stuff need to happen for the new major, cannot be done afterwards.